### PR TITLE
CP-942 Document FluxComponent and Store

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Issue
In preparation of OSS, we need to fill out the source documentation.

## Changes
Docstrings added for..
- `FluxComponent`
- `Store`
- The main `w_flux` library

> **Docs only change.**

## Areas of Regression
n/a

## Testing
n/a

## Code Review
@trentgrover-wf
@maxwellpeterson-wf 
@dustinlessard-wf
@jayudey-wf

### CODE REVIEW

File: lib/src/component.dart [hunk 1]
```diff
@@ -20,13 +20,47 @@ import 'package:react/react.dart' as react;
 
 import './store.dart';
 
+/// FluxComponents are responsible for rendering application view and turning
+/// user interactions and events into [Action]s. FluxComponents can use data
+/// from one or many [Store] instances to define the resulting component.
+///
+///
```
Comment (maxwellpeterson-wf) [inline lib/src/component.dart]: Extra lines?

File: lib/src/component.dart [hunk 2]
```diff
@@ -20,13 +20,47 @@ import 'package:react/react.dart' as react;
 
 import './store.dart';
 
+/// FluxComponents are responsible for rendering application view and turning
```
Comment (dustinlessard-wf) [inline lib/src/component.dart]: *views ?

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by dustinlessard-wf** [comment]
+1

**Comment by dustinlessard-wf** [comment]
+1

File: lib/src/component.dart [hunk 3]
```diff
@@ -43,6 +76,20 @@ abstract class FluxComponent<ActionsT, StoresT> extends react.Component {
...
 
+  /// Define the list of [Store] instances that this component should listen to.
+  /// When any of the returned [Store]s update their state, this component will
+  /// redraw.
+  ///
+  /// If [store] is of type [Store] (in other words, if this component has a
+  /// single Store passed in), this will return a list with said store as the
+  /// only element by default. Otherwise, an empty list is returned.
+  ///
+  /// This method should be overridden if [store] is actually a composite object
```
Comment (trentgrover-wf) [inline lib/src/component.dart]: rephrase for clarity sake:
```
If [store] is actually a composite object with multiple stores, this method should be overridden to return...
```

File: lib/src/component.dart [hunk 4]
```diff
@@ -51,10 +98,20 @@ abstract class FluxComponent<ActionsT, StoresT> extends react.Component {
     }
   }
 
+  /// If you need more fine-grained control over listening to stores, override
```
Comment (trentgrover-wf) [inline lib/src/component.dart]: `...fine-grained control over store trigger handling...`

File: lib/src/component.dart [hunk 5]
```diff
@@ -51,10 +98,20 @@ abstract class FluxComponent<ActionsT, StoresT> extends react.Component {
...
+  /// this method to return a Map of stores to handlers. Whenever a store in
+  /// the returned map triggers, the respective handler will be called.
+  ///
+  /// If possible, [redrawOn] should be used instead of this in order to avoid
```
Comment (trentgrover-wf) [inline lib/src/component.dart]: should probably note that `getStoreHandlers` have precedence over `redrawOn` handling

File: lib/src/store.dart [hunk 6]
```diff
@@ -18,10 +18,39 @@ import 'dart:async';
 
 import 'package:w_flux/src/action.dart';
 
+/// A `Store` is a repository and manager of app state. This class should be
+/// extended to fit the needs of your application and its data. The number and
+/// hierarchy of stores is dependent upon the state management needs of your
+/// application.
+///
+/// General guidelines with respect to a `Store`s data:
```
Comment (trentgrover-wf) [inline lib/src/store.dart]: missing the ' in `Store`'s

File: lib/src/store.dart [hunk 7]
```diff
@@ -18,10 +18,39 @@ import 'dart:async';
...
+/// - A `Store`'s data should not be exposed for direct mutation.
+/// - A `Store`'s data should be mutated by itself in response to [Action]s.
```
Comment (trentgrover-wf) [inline lib/src/store.dart]: `by itself` -> `internally`?

File: lib/src/store.dart [hunk 8]
```diff
@@ -34,10 +63,22 @@ class Store {
     }
   }
 
+  /// Trigger a "data updated" event. All registered listeners of this `Store`
+  /// will be receive the event, at which point they can use the latest data
```
Comment (trentgrover-wf) [inline lib/src/store.dart]: extra `be`

File: lib/src/store.dart [hunk 9]
```diff
@@ -34,10 +63,22 @@ class Store {
...
+  /// from this `Store` as necessary.
+  ///
+  /// This should be called whenever this `Store`s data has finished mutating in
```
Comment (trentgrover-wf) [inline lib/src/store.dart]: missing ' in `Store`'s

File: lib/src/store.dart [hunk 10]
```diff
@@ -34,10 +63,22 @@ class Store {
...
+  /// response to an action.
   void trigger() {
     _streamController.add(this);
   }
 
+  /// A convenience method for listening to an [action] and triggering
+  /// automatically once the callback for said action has completed.
+  ///
+  /// If [onAction] is provided, it will be called for every time [action] is
```
Comment (trentgrover-wf) [inline lib/src/store.dart]: nix the `for`?

File: lib/w_flux.dart [hunk 11]
```diff
@@ -12,6 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// The w_flux library implements a uni-directional data flow pattern comprised
+/// of [Action]s, [Store]s, and [FluxComponent]s.
+///
+/// - [Action]s initiate mutation of app data that reside in [Store]s.
```
Comment (trentgrover-wf) [inline lib/w_flux.dart]: `resides`

Comment (evanweible-wf) [inline lib/w_flux.dart]: I guess I was interpreting "data" as plural here. English is awful

**Comment by evanweible-wf** [comment]
@trentgrover-wf comments addressed

**Comment by trentgrover-wf** [comment]
+1
@jayudey-wf ready for merge (inline docs change only)

**Comment by jayudey-wf** [comment]
QA +1, documentation change only

Merging into master

**Comment by jayudey-wf** [comment]
@rosie run_merge_script

**Comment by jayudey-wf** [comment]
@rosie run_merge_script

**Pull request merged**